### PR TITLE
Bugfix/KBDEV-1156 Failure to Change User Permissions in Admin

### PR DESCRIPTION
**Pull request by bnguyen-bcgsc**
- Fix bug where admin making adjustments to users result in 'Failed casting email: Buffer is not defined' error
- Fix random typescript errors
- Update FormContext and useSchemaForm for consistency of properties

### CODE REVIEW

File: src/static/graphkb-env-config.js [hunk 1]
```diff
@@ -1,5 +1,3 @@
-// default config.js file, gets replaced at run-time by env-scripts
```
Comment (mathieulemieux) [inline src/static/graphkb-env-config.js:1]: My understanding is that updates to that file should usually not be commited, am I wrong?

Comment (bnguyen-bcgsc) [reply src/static/graphkb-env-config.js:1]: Pinging @elewis2 . I can also add this file to gitignore

Comment (mathieulemieux) [reply src/static/graphkb-env-config.js:1]: But we want the file to be in the repo

Comment (bnguyen-bcgsc) [reply src/static/graphkb-env-config.js:1]: ohh I see so we just have to ignore the changes at run-time to it right?

Comment (mathieulemieux) [reply src/static/graphkb-env-config.js:1]: Yep, I guess so.

**Review by mathieulemieux — APPROVED** [review]

**Pull request merged**